### ISSUE TITLE
RESPONSE refs can not be internal references. Why?

### ISSUE THREAD

**Issue by ChrisTitos** [swagger-api/swagger-core#1388]
Hi,

I'm having troubles using the latest swagger-codegen. I get this error:
```
RESPONSE refs can not be internal references 
(through reference chain: io.swagger.models.RefResponse["$ref"]) 
(through reference chain: io.swagger.models.Path["get"]->io.swagger.models.Operation["responses"]->java.util.LinkedHashMap["404"]) 
(through reference chain: io.swagger.models.Swagger["paths"]->java.util.LinkedHashMap["/companies/{id}"])
```

This exception is thrown by swagger-core here: 
https://github.com/swagger-api/swagger-core/blob/master/modules/swagger-models/src/main/java/io/swagger/models/refs/GenericRef.java#L37

In the comment it says "PATH AND RESPONSE refs  can only be URL or RELATIVE", but my question is, where does it say this in the spec? The [Responses object](http://swagger.io/specification/#responsesObject) doesn't mention any limitations to a reference object.

I might be missing something, but this exception confused me (especially because the online codegen doesn't complain)

**Comment by webron** [comment]
When you say latest swagger-codegen, do you mean the master branch or develop_2.0?

**Comment by ChrisTitos** [comment]
At first I thought it was both master and develop, but after doing a clean first, now it only happens in develop_2.0.
I also get the exception when using swagger-parser v1.0.10

**Comment by ChrisTitos** [comment]
The develop_2.0 branch has been merged into codegen master, so now I also get the exception when using the master branch.
But like I said, this exception shouldn't even be there right? The spec says nothing about what kind of reference is allowed or not (or am I misinterpreting something?).

**Comment by ePaul** [comment]
This error message seems to have been introduced in #1271 (commit 59e6b10).
I didn't find any justification for this in the discussion of the Pull request, nor in the commit comment (which doesn't say much at all).

**Comment by ePaul** [comment]
Related to this, there seems to be no `responses` property of the Swagger object. Maybe this was the reason?

**Comment by fehguy** [comment]
this has been added and will be available in 1.5.4
